backend/utils/message_utils.py

```diff
@@ -103,6 +103,59 @@ async def get_room_messages(
     return response
 
 
+async def get_room_messages_for_threads(
```

Comment:
this function will just return the same result as `get_room_messages` excluding pagination. If you want to write a utility function that returns threaded messages, then the logic should be different from that go get room messages.